src/acl/ConnMark.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+/* DEBUG: section 28    Access Control */
+
+#include "squid.h"
+#include "acl/ConnMark.h"
+#include "acl/FilledChecklist.h"
+#include "cache_cf.h"
+#include "client_side.h"
+#include "Debug.h"
+#include "http/Stream.h"
+#include "HttpReply.h"
+#include "HttpRequest.h"
+#include "Parsing.h"
+#include "sbuf/Stream.h"
+
+#include <string>
+
+bool
+Acl::ConnMark::empty () const
+{
+    return false;
+}
+
+inline std::ostream &
+operator <<(std::ostream &os, const std::pair<nfmark_t,nfmark_t> connmark)
+{
+    using namespace std;
+    ios_base::fmtflags oldFlags = os.flags();
+
+    os << hex << showbase << connmark.first;
+    if (connmark.second != 0xffffffff) {
+        os << "/" << connmark.second;
+    }
+
+    os.setf(oldFlags);
+    return os;
+}
+
+nfmark_t
+Acl::ConnMark::getNumber(Parser::Tokenizer &tokenizer, const SBuf &token)
+{
+    int64_t number;
+
+    if (!tokenizer.int64(number, 0, false)) {
+        throw TexcHere(ToSBuf("acl ", typeString(), ": invalid value '", tokenizer.buf(), "' in ", token));
+    }
+
+    if (number > std::numeric_limits<nfmark_t>::max()) {
+        throw TexcHere(ToSBuf("acl ", typeString(), ": number ", number, " is too big"));
+    }
+
+    return static_cast<nfmark_t>(number);
+}
+
+void
+Acl::ConnMark::parse()
+{
+    while (const char *t = ConfigParser::strtokFile()) {
+        SBuf token(t);
+        Parser::Tokenizer tokenizer(token);
+        nfmark_t mark = 0;
+        nfmark_t mask = 0xffffffff;
+
+        mark = getNumber(tokenizer, token);
+        if (tokenizer.skip('/')) {
+            mask = getNumber(tokenizer, token);
+        }
+
+        if (!tokenizer.atEnd()) {
+            throw TexcHere(ToSBuf("acl ", typeString(), ": trailing garbage in ", token));
+        }
+
+        std::pair<nfmark_t, nfmark_t> connmark(mark, mask);
+        marks.insert(connmark);
+        debugs(28, 7, "mark '" << connmark << "'");
+    }
+
+    if (marks.empty()) {
+        throw TexcHere(ToSBuf("acl ", typeString(), " requires at least one mark"));
+    }
+}
+
+int
+Acl::ConnMark::match(ACLChecklist *cl)
+{
+    const auto *checklist = Filled(cl);
+    nfmark_t connMark = checklist->conn()->clientConnection->nfmark;
```

Comment:
If possible, make connMark constant. You can use `const auto` here AFAICT.

Also, please use either `connMark` or `connmark` for consistency sake. I recommend the former.